src/rust_ir.rs

```diff
@@ -18,22 +19,22 @@ pub mod lowering;
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Program {
     /// From type-name to item-id. Used during lowering only.
-    crate type_ids: BTreeMap<Identifier, ItemId>,
+    crate type_ids: BTreeMap<Identifier, TypeId>,
 
     /// For each struct/trait:
-    crate type_kinds: BTreeMap<ItemId, TypeKind>,
+    crate type_kinds: BTreeMap<TypeKindId, TypeKind>,
 
     /// For each struct:
-    crate struct_data: BTreeMap<ItemId, StructDatum>,
+    crate struct_data: BTreeMap<StructId, StructDatum>,
```

Comment:
Given that per-type ids are now dense, one *can* use a `Vec` or arena here, instead of `BTreeMap`. I don't think we should change this though: my understanding that we are going to refactor out this struct to a set of queries anyway.

Reply:
Yeah, leave it